packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,173 @@
+          nestedStacks: {
+            "GrandChildStack": {
+              templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+      },
+    });
+
+    const grandChildTemplate = parent.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack').includedTemplate;
+    const bucket = grandChildTemplate.getResource('BucketImport') as s3.CfnBucket;
+
+    bucket.bucketName = 'modified-bucket-name';
+
+    const grandChildStack = parent.getNestedStack('ChildStack').includedTemplate.getNestedStack('GrandChildStack').stack;
+    expect(grandChildStack).toHaveResource('AWS::S3::Bucket', { BucketName: 'modified-bucket-name' });
+  });
+
+  test('templates with nested stacks that do not exist in the nested stacks property have those stacks ignored', () => {
+    new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+    });
+
+    expect(stack).toMatchTemplate(loadTestFileToJsObject('parent-export-stack.json'));
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that does not exist in the template', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('FakeStack');
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but is not a Stack', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('ChildStack').includedTemplate.getNestedStack('BucketImport');
+    }).toThrow(/Resource with logical ID 'BucketImport' is not a CloudFormation Stack/);
+  });
+
+  test('getNestedStack() throws an exception when getting a resource that exists in the template, but was not specified in the props', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+        },
+      },
+    });
+
+    expect(() => {
+      parentTemplate.getNestedStack('AnotherChildStack').includedTemplate.getNestedStack('BucketImport');
+    }).toThrow(/Nested Stack with logical ID 'AnotherChildStack' was not specified in the CfnInclude props/);
```

Comment:
```suggestion
    }).toThrow(/Nested Stack 'AnotherChildStack' was not included in the nestedStacks property when including the parent stack, and so cannot be retrieved from it/);
```

Best I could come up with, but basically, this error needs to be more actionable.